hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -78,15 +78,20 @@ object Emit {
     }
     val emitter = new Emit[C](ctx, fb.ecb)
     if (ir.typ == TVoid) {
-      val cb = EmitCodeBuilder(mb)
-      emitter.emitVoid(cb, ir, mb, Env.empty, container)
-      fb.emit(cb.result())
+      fb.emitWithBuilder { cb =>
+        emitter.emitVoid(cb, ir, mb, Env.empty, container)
+        Code._empty
+      }
     } else {
-      val triplet = emitter.emit(ir, mb, Env.empty, container)
-      val ti = typeToTypeInfo(ir.pType)
-      fb.emit(Code(triplet.setup, triplet.m.muxAny(
-        Code._throwAny[RuntimeException](Code.newInstance[RuntimeException, String]("cannot return empty"))(ti),
-        triplet.v)))
+      fb.emitWithBuilder { cb =>
+        // FIXME (use IEmitCode.consume)
```

Comment:
I see why this didn't work. Maybe you want something like
```
  def handle(cb: EmitCodeBuilder)(ifMissing: => Unit): PCode = {
    cb.define(Lmissing)
    ifMissing
    cb.define(Lpresent)
    pc
  }
```
?
Then you could write
```
fb.emitWithBuilder { cb =>
  emitter.emitI(ir, cb, Env.empty, container).handle {
    cb._throw[RuntimeException](
      Code.newInstance[RuntimeException, String]("cannot return empty"))
  }
}
```

Reply:
This sounds right. Thanks!